packages/hash/frontend/src/blocks/page/createSuggester/BlockSuggester.tsx

```diff
@@ -1,18 +1,36 @@
 import { BlockVariant } from "blockprotocol";
-import { BlockMeta } from "@hashintel/hash-shared/blockMeta";
 import { useMemo, VFC } from "react";
 import { tw } from "twind";
 
-import { useBlocksMeta } from "../../blocksMeta";
 import { fuzzySearchBy } from "./fuzzySearchBy";
 import { Suggester } from "./Suggester";
+import { UserBlock, useUserBlocks } from "../../userBlocks";
+import { useFilteredBlocks } from "./useFilteredBlocks";
 
 export interface BlockSuggesterProps {
   search?: string;
-  onChange(variant: BlockVariant, block: BlockMeta): void;
+  onChange(variant: BlockVariant, block: UserBlock): void;
   className?: string;
 }
 
+// @todo remove this when API returns actual icon URL
+export const getVariantIcon = (option: {
```

Comment:
This is a temporary hack to generate icon for a block or a variant which doesn't have an absolute URL. 

I'll create a task for this if we want the blocks API to return proper URLs.

Reply:
Created a task for this here: https://app.asana.com/0/1200211978612931/1201988634400654/f

Reply:
I think this is another example of why you need to generate the `componentId` at the time you fetch the blocks. This is needless complexity.